route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -1,41 +1,44 @@
-use crate::link::{Link, PacketStream, TokioRunnable};
+use crate::link::{Link, LinkBuilder, PacketStream};
 use futures::{Async, Future, Poll};
 
 /// Link that drops all packets ingressed.
 #[derive(Default)]
-pub struct BlackHoleLink<Packet: Sized> {
-    hole: Option<BlackHole<Packet>>,
+pub struct BlackHoleLinkBuilder<Packet: Sized> {
+    in_streams: Option<Vec<PacketStream<Packet>>>,
 }
 
-impl<Packet> BlackHoleLink<Packet> {
+impl<Packet> BlackHoleLinkBuilder<Packet> {
     pub fn new() -> Self {
-        BlackHoleLink { hole: None }
+        BlackHoleLinkBuilder { in_streams: None }
     }
 }
 
-impl<Packet: Sized + 'static> Link<Packet, ()> for BlackHoleLink<Packet> {
-    fn ingressors(&self, ingress_streams: Vec<PacketStream<Packet>>) -> Self {
-        BlackHoleLink {
-            hole: Some(BlackHole::new(ingress_streams)),
+impl<Packet: Sized + 'static> LinkBuilder<Packet, ()> for BlackHoleLinkBuilder<Packet> {
+    fn ingressors(self, ingress_streams: Vec<PacketStream<Packet>>) -> Self {
+        BlackHoleLinkBuilder {
+            in_streams: Some(ingress_streams),
```

Comment:
With multiple builder functions, the pattern of "accumulate all of your state, then initialize in `build_link`" became clearer, so `BlackHoleLink` is updated to reflect that.